pkg/endpointmanager/manager.go

```diff
@@ -130,6 +130,26 @@ func (mgr *endpointManager) WithPeriodicEndpointGC(ctx context.Context, checkHea
 	return mgr
 }
 
+// WithPeriodicEndpointRegeneration configures the EndpointManager to regenerate all (policy and configuration) state
+// of endpoints periodically.
+func (mgr *endpointManager) WithPeriodicEndpointRegeneration(ctx context.Context, interval time.Duration) *endpointManager {
+	mgr.controllers.UpdateController("endpoint-periodic-regeneration",
+		controller.ControllerParams{
+			Group: endpointGCControllerGroup,
+			DoFunc: func(_ context.Context) error {
+				wg := mgr.RegenerateAllEndpoints(&regeneration.ExternalRegenerationMetadata{
+					Reason:            "periodic endpoint regeneration",
+					RegenerationLevel: regeneration.RegenerateWithoutDatapath,
+				})
+				wg.Wait()
+				return nil
+			},
+			RunInterval: interval,
```

Comment:
I think if this is zero this will still do one endpoint regenerate when `UpdateController` is called, is that intended or should we return early if our interval is zero?

Reply:
We *also* use this controller to trigger regeneration as needed (i.e. when Envoy configurations are updated) so we can't skip setting it up when the interval is 0.

In the unlikely case that an end-user disables this controller, we will trigger one solitary regeneration on startup. This will do nothing, as there are no endpoints yet. I think this is OK :-).